x-pack/plugins/security/public/authentication/login/components/login_form/login_form.tsx

```diff
@@ -279,12 +390,10 @@ export class LoginForm extends Component<Props, State> {
     this.validator.enableValidation();
 
     const { username, password } = this.state;
-    const result = this.validator.validateForLogin(username, password);
-    if (result.isInvalid) {
-      this.setState({ formError: result });
-      return;
-    } else {
-      this.setState({ formError: null });
+    if (this.validator.validateForLogin(username, password).isInvalid) {
+      // Since validation is enabled now, we should ask React to re-render form and display
+      // validation error messages if any.
+      return this.forceUpdate();
```

Comment:
We need a new pattern for this across the security screens...we abuse state in other components to achieve this, but your change to use `forceUpdate` is at least more explicit. You can ignore this comment, I'm just thinking out loud for the future

Reply:
Yeah, totally agree.